mp/src/game/server/momentum/mom_player.cpp

```diff
@@ -1879,9 +1879,17 @@ void CMomentumPlayer::TimerCommand_Restart(int track)
     {
         if (!g_pSavelocSystem->TeleportToStartMark(START_MARK, track))
         {
-            // Don't set angles if still in start zone.
-            QAngle ang = pStart->GetLookAngles();
-            ManualTeleport(&pStart->GetRestartPosition(), (pStart->HasLookAngles() ? &ang : nullptr), &vec3_origin);
+            const auto pZoneTeleDest = pStart->GetTeleDest();
+            if (pZoneTeleDest)
+            {
+                ManualTeleport(&pZoneTeleDest->GetAbsOrigin(), &pZoneTeleDest->GetAbsAngles(), &vec3_origin);
+            }
+            else
+            {
+                // Don't set angles if still in start zone.
+                QAngle ang = pStart->GetLookAngles();
+                ManualTeleport(&pStart->GetRestartPosition(), pStart->HasLookAngles() ? &ang : nullptr, &vec3_origin);
+            }
```

Comment:
The only thing changing here is what's passed in for target dest and angles. Could be good to instead make the logic about selecting the position and angles instead of calling ManualTeleport multiple times:

```suggestion
            Vector *pTargetDest = nullptr;
            QAngle *pTargetAng = nullptr;
            const auto pZoneTeleDest = pStart->GetTeleDest();
            if (pZoneTeleDest)
            {
                pTargetDest = pZoneTeleDest->GetAbsOrigin();
                pTargetAng = pZoneTeleDest->GetAbsAngles();
            }
            else
            {
                pTargetDest = &pStart->GetRestartPosition();
                pTargetAng = pStart->HasLookAngles() ? &pStart->GetLookAngles() : nullptr;
            }

            ManualTeleport(pTargetDest, pTargetAng, &vec3_origin);
```

Reply:
Did this for restart stage as well.